src/Application/Digital/Commands/ShareEntry.cs

```diff
@@ -0,0 +1,178 @@
+                && x.EmployeeId == request.CurrentUser.Id
+                && x.AllowedOperations.Contains(EntryOperation.ChangePermission.ToString()));
+            
+            if (entry.Owner!.Id != request.CurrentUser.Id && !hasModifyPermission)
+            {
+                throw new UnauthorizedAccessException("You are not allow to change permission of this entry.");
+            }
+            
+            var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == request.UserId, cancellationToken);
+
+            if (user is null)
+            {
+                throw new KeyNotFoundException("User does not exist.");
+            }
+
+            if (user.Id == entry.OwnerId)
+            {
+                throw new ConflictException("You can not modify owner's permission.");
+            }
+
+            if (request.ExpiryDate < _dateTimeProvider.DateTimeNow)
+            {
+                throw new ConflictException("Expiry date cannot be in the past.");
+            }
+
+            var allowOperations = GenerateAllowOperations(request, entry.IsDirectory);
+            
+            await GrantPermission(entry, user, allowOperations, request.ExpiryDate, cancellationToken);
+
+            if (entry.IsDirectory)
+            {
+                // Update permissions for child Entries
+                var path = entry.Path.Equals("/") ? (entry.Path + entry.Name) : (entry.Path + "/" + entry.Name);
+                var childEntries = _context.Entries
+                    .Include(x => x.Uploader)
+                    .Include(x => x.Owner)
+                    .Where(x => x.Id != entry.Id 
+                                && x.Path.StartsWith(path)
+                                && x.OwnerId == entry.OwnerId)
+                    .ToList();
+                foreach (var childEntry in childEntries)
+                {
+                    var childAllowOperations = GenerateAllowOperations(request, childEntry.IsDirectory);
+                    await GrantPermission(childEntry, user, childAllowOperations, request.ExpiryDate, cancellationToken);
+                }
+            }
+
+            var result = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == request.EntryId
+                    && x.EmployeeId == request.UserId
+                , cancellationToken);
+            
+            await _context.SaveChangesAsync(cancellationToken);
+            return _mapper.Map<EntryPermissionDto>(result);
+        }
+
+        private async Task GrantPermission(
+            Entry entry,
+            User user,
+            string allowOperations,
+            DateTime expiryDate,
+            CancellationToken cancellationToken)
+        {
+            var existedPermission = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == entry.Id
+                    && x.EmployeeId == user.Id
+                , cancellationToken);
```

Comment:
this does not need to be down-line

Reply:
uhmm...no? ok,,,

Reply:
Fixed at https://github.com/ProFile-org/ProFile-BE/pull/336/commits/878b9659ef059cf753c54d4198d1dfcf167e90da